src/pages/Home/Home.js

```diff
@@ -1,38 +1,55 @@
 import React, { useState, useEffect } from 'react';
-import { useSelector, useDispatch } from 'react-redux';
+import { useDispatch } from 'react-redux';
 import { useParams, useSearchParams } from 'react-router-dom';
 import ErrorBoundary from '../../hoc/ErrorBoundary/ErrorBoundary';
 import Header from '../../views/Header/Header';
 import MainContent from '../../views/MainContent/MainContent';
 import Footer from '../../views/Footer/Footer';
-import { selectMovieById, fetchMovies } from '../../store/moviesSlice';
+import { fetchMovies } from '../../store/moviesSlice';
 import { useURLSearchParams } from '../../hooks/index';
+import { getMovieByIdRequest } from '../../api/index';
 
 export default function Home() {
   const dispatch = useDispatch();
   const [searchParams, setSearchParams] = useSearchParams();
   const [showMovieDetails, setShowMovieDetails] = useState(false);
-  const [movieId, setMovieId] = useState();
+  const [movie, setMovie] = useState();
 
-  const movie = useSelector((state) => selectMovieById(state, movieId));
+  const sortBy = useURLSearchParams().get('sortBy') || 'vote_average';
+  const genre = useURLSearchParams().get('genre') || '';
+  const movieId = useURLSearchParams().get('movie') || '';
+  const { searchQuery: search } = useParams();
 
   const onMovieClick = (movieId) => {
     searchParams.set('movie', movieId);
     setSearchParams(searchParams);
 
-    setShowMovieDetails(true);
-    setMovieId(movieId);
     window.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
   };
 
-  const sortBy = useURLSearchParams().get('sortBy') || 'vote_average';
-  const genre = useURLSearchParams().get('genre') || '';
-  const { searchQuery: search } = useParams();
-
   useEffect(() => {
     dispatch(fetchMovies({ sortBy, genre, search }));
   }, [dispatch, sortBy, genre, search]);
 
+  useEffect(() => {
+    const fetchMovieById = async (id) => {
+      const response = await getMovieByIdRequest(id);
+
+      if (response.status === 200) {
+        setMovie(response.data);
+        setShowMovieDetails(true);
+      } else {
+        // eslint-disable-next-line no-alert
+        alert('Movie not found');
+        setShowMovieDetails(false);
+      }
+    };
+
+    if (movieId) {
+      fetchMovieById(movieId);
+    }
+  }, [movieId]);
```

Comment:
Can we use some other approach here or is it ok?)

Reply:
Yeah, this is the right approach with a bit of strange complexity of React side effects. No need to change anything here

Reply:
A better approach would be to use react-query package, but I think it's out of the scope of this task